pkg/storage/storage.go

```diff
@@ -142,6 +151,52 @@ func (s *Storage) History(name string) ([]*rspb.Release, error) {
 	return s.Driver.Query(map[string]string{"NAME": name, "OWNER": "TILLER"})
 }
 
+// removeLeastRecent removes items from history until the lengh number of releases
+// does not exceed max.
+//
+// We allow max to be set explicitly so that calling functions can "make space"
+// for the new records they are going to write.
+func (s *Storage) removeLeastRecent(name string, max int) error {
+	if max <= 0 {
```

Comment:
- [ ] Did you mean just less than zero rather than less than _or equal to_ zero?  
  If the maximum history count is one, then at line 60 above we'll call `removeLeastRecent` with an argument of zero for the "max" parameter, which will then do nothing. If we have even just one release stored in the history, and we want at most one, then even when called with a count of zero here we need to remove one release to bring the historical count down to zero in anticipation of adding one new record.